nexuse2e-core/src/main/java/org/nexuse2e/service/http/HttpReceiverService.java

```diff
@@ -99,6 +111,23 @@ public ModelAndView handleRequest(HttpServletRequest request, HttpServletRespons
                 return null;
             }
 
+            Boolean basicAuthEnabled = getParameter(BASIC_AUTH_ENABLED);
+            if(basicAuthEnabled != null && basicAuthEnabled) {
+                String authHeader = request.getHeader("Authorization");
+                if(StringUtils.isBlank(authHeader)) {
+                    response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+                    return null;
+                }
+                String basicAuthUsername = getParameter(BASIC_AUTH_NAME);
+                String basicAuthPassword = getParameter(BASIC_AUTH_PASSWORD);
+                byte[] tokenBytes = Base64.encodeBase64((basicAuthUsername+":"+basicAuthPassword).getBytes(StandardCharsets.UTF_8));
+                String expected = "Basic "+new String(tokenBytes,StandardCharsets.UTF_8);
+                if(!Objects.equals(expected,authHeader)) {
+                    response.setStatus(HttpServletResponse.SC_FORBIDDEN);
```

Comment:
The RFC says it should return 401 Unauthorized and not 403 Forbidden. https://www.rfc-editor.org/rfc/rfc7617